planet.js

```diff
@@ -0,0 +1,57 @@
+let nameH1;
+let filmsDiv;
+let planetDiv;
+let charactersDiv;
+const baseUrl = `https://swapi2.azurewebsites.net/api`;
+
+// Runs on page load
+addEventListener('DOMContentLoaded', () => {
+  nameH1 = document.querySelector('h1#name');
+  charactersUl = document.querySelector('#characters>ul');
+  filmsUl = document.querySelector('#films>ul');
+
+  const sp = new URLSearchParams(window.location.search)
+  const id = sp.get('id')
+  getPlanet(id)
+});
+
+async function getPlanet(id) {
+  let planet;
+  try {
+    planet = await fetchPlanet(id)
+    planet.films = await fetchFilms(planet)
+  }
+  catch (ex) {
+    console.error(`Error reading planet ${id} data.`, ex.message);
+  }
+  renderPlanet(planet);
+
+}
+async function fetchPlanet(id) {
+  let planetUrl = `${baseUrl}/planets/${id}`;
+  return await fetch(planetUrl)
+    .then(res => res.json())
+}
+
+async function fetchCharacters(planet) {
+  const url = `${baseUrl}/planets/${planet?.id}/characters`;
+  const characters = await fetch(url)
+    .then(res => res.json())
+  return characters;
+}
+
+async function fetchFilms(planet) {
+  const url = `${baseUrl}/planets/${planet?.id}/films`;
+  const films = await fetch(url)
+    .then(res => res.json())
+  return films;
+}
+
+const renderPlanet = planet => {
+  document.title = `SWAPI - ${planet?.name}`;  // Just to make the browser tab say planet name
+  nameH1.textContent = planet?.name;
+  const charactersLis = planet?.characters?.map(character => `<li><a href="/character.html?id=${character.id}"</li>`)
```

Comment:
const charactersLis = planet?.characters?.map(character => `<li><a href="/character.html?id=${character.id}">${character.name}</li>`)

Missing the end of the anchor tag and the string literal --  add: >${character.name}

Reply:
Changes Done